renderdoc/driver/d3d12/d3d12_shaderdebug.cpp

```diff
@@ -324,6 +324,91 @@ void D3D12DebugManager::CreateShaderGlobalState(ShaderDebug::GlobalState &global
   global.PopulateGroupshared(dxbc->GetDXBCByteCode());
 }
 
+void GatherConstantBuffers(WrappedID3D12Device *pDevice, DXBC::ShaderType shaderType,
+                           const D3D12RenderState::RootSignature &rootsig,
+                           bytebuf cbufData[D3D12_COMMONSHADER_CONSTANT_BUFFER_API_SLOT_COUNT])
```

Comment:
I don't understand why `D3D12_COMMONSHADER_CONSTANT_BUFFER_API_SLOT_COUNT` exists, I think a lot of those constants were copy-pasted from D3D11 and renamed, given that `D3D12_SHADER_MAX_INTERFACES` exists but D3D12 doesn't support shader interfaces at all...

Either way this limit only applies to 'legacy' SM5.0 shaders that are limited to D3D11 compatibility. SM5.1 shaders can have any register value. E.g.:

```
cbuffer b : register(b100000000)
{
  float4 a;
};

float4 main() : SV_Target0
{
  return a;
}
```

produces

```
ps_5_1
dcl_globalFlags refactoringAllowed
dcl_constantbuffer CB0[100000000:100000000][1], immediateIndexed, space=0
dcl_output o0.xyzw
mov o0.xyzw, CB0[100000000][0].xyzw
ret
```

This also doesn't account for register spaces, which can also be arbitrary numbers and can allow the same register index to exist in multiple space with different cbuffers.

Bottom line is that this will have to be a map from `{space, register}` to data.